code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -258,6 +250,12 @@
 		newtonian_move(get_dir(target, user))
 	A.original = target
 	A.fire()
+	if(casingtype)
+		var/obj/item/ammo_casing/casing = new casingtype(get_turf(src))
+		if(casing.BB)
+			qdel(casing.BB)

```

Comment:
This doesn't delete the `casing` object that was just created, but rather one of its vars
Which seems kinda strange to me; couldn't there be an arg in `New()` or something else to avoid that?